### PR TITLE
HUD: Auto-scaling, animated images and custom font size

### PR DESCRIPTION
**Pull request by kilbith**
Some enhancements to the HUD:

1. HUD elements are scaled automatically according to the screen width.
2. Playing animated images.
3. Setting a custom font size.

Known issue: a new client crashes when joining an out-of-date server.

### CODE REVIEW

**Comment by Calinou** [comment]
How does `hud_grid_resolution` work in conjuction with `hud_scaling` and `gui_scaling`? Also, why is it a float rather than an integer value?

File: src/texture.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#pragma once
```
Comment (nerzhul) [inline src/texture.h]: missing license headers, and as it's purely client side cpp/header please move them to client folder

**Review by nerzhul — CHANGES_REQUESTED** [review]

File: src/network/clientpackethandler.cpp [hunk 2]
```diff
@@ -1116,6 +1117,8 @@ void Client::handleCommand_HudAdd(NetworkPacket* pkt)
 		*pkt >> size;
 	} catch(SerializationError &e) {};
 
+	*pkt >> font_size;
```
Comment (nerzhul) [inline src/network/clientpackethandler.cpp]: move that into the try catch block

**Comment by Wuzzy2** [comment]
Will this fix https://github.com/minetest/minetest/issues/6296?

File: src/network/clientpackethandler.cpp (hunk 2 above)
Comment (SmallJoker) [reply src/network/clientpackethandler.cpp]: Plus the old try/catch blocks can be removed, as these params are guaranteed to be sent after the backwards compat drop.

**Comment by kilbith** [comment]
Doomed to rot there (how surprising). Closing.

**Comment by Wuzzy2** [comment]
Nooo, please don't close this. Have nerzhul's comments been considered?

**Pull request closed without merging**